### PR TITLE
[16.0][FIX] intrastat_product: Do not display missing VAT number error when…

### PR DESCRIPTION
**Pull request by aritzolea**
… fiscal position is B2C

### CODE REVIEW

File: intrastat_product/tests/common_sale.py [hunk 1]
```diff
@@ -55,6 +55,9 @@ def _init_customer(cls, vals=None):
         if vals is not None:
             values.update(vals)
         cls.customer = cls.partner_obj.create(values)
+        values.pop("vat")
+        values["property_account_position_id"] = cls.position_b2c.id
+        cls.customer_no_vat = cls.partner_obj.create(values)
```
Comment (ljsalvatierra-factorlibre) [inline intrastat_product/tests/common_sale.py:60]: Another way of doing it.

```python
cls.customer_no_vat = cls.customer.copy(
    {
        "vat": False,
        "property_account_position_id": cls.position_b2c.id,
    }
)
```

**Review by ljsalvatierra-factorlibre — APPROVED** [review]
LGTM

**Review by MayteGLC — APPROVED** [review]
LGTM

File: intrastat_product/models/intrastat_product_declaration.py [hunk 2]
```diff
@@ -492,6 +492,9 @@ def _get_partner_and_warn_vat(self, inv_line, notedict):
                         notedict["inv_origin"]
                     )
             else:
+                if inv.fiscal_position_id:
+                    if inv.fiscal_position_id.intrastat == "b2c":
+                        return partner
```
Comment (pedrobaeza) [inline intrastat_product/models/intrastat_product_declaration.py]:
```suggestion
            elif inv.fiscal_position_id.intrastat != "b2c":
```

Comment (aritzolea) [reply intrastat_product/models/intrastat_product_declaration.py]: Modified!

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge patch

**Pull request merged**